piptools/repositories/pypi.py

```diff
@@ -156,50 +211,21 @@ def get_dependencies(self, ireq):
             if not os.path.isdir(self._wheel_download_dir):
                 os.makedirs(self._wheel_download_dir)
 
+            wheel_cache = WheelCache(CACHE_DIR, self.pip_options.format_control)
+            prev_tracker = os.environ.get('PIP_REQ_TRACKER')
             try:
-                # Pip < 9 and below
-                reqset = RequirementSet(
-                    self.build_dir,
-                    self.source_dir,
-                    download_dir=download_dir,
-                    wheel_download_dir=self._wheel_download_dir,
-                    session=self.session,
-                    wheel_cache=self.wheel_cache,
-                )
-                self._dependencies_cache[ireq] = reqset._prepare_file(
-                    self.finder,
-                    ireq
-                )
-            except TypeError:
-                # Pip >= 10 (new resolver!)
-                preparer = RequirementPreparer(
-                    build_dir=self.build_dir,
-                    src_dir=self.source_dir,
-                    download_dir=download_dir,
-                    wheel_download_dir=self._wheel_download_dir,
-                    progress_bar='off',
-                    build_isolation=False
-                )
-                reqset = RequirementSet()
-                ireq.is_direct = True
-                reqset.add_requirement(ireq)
-                self.resolver = PipResolver(
-                    preparer=preparer,
-                    finder=self.finder,
-                    session=self.session,
-                    upgrade_strategy="to-satisfy-only",
-                    force_reinstall=False,
-                    ignore_dependencies=False,
-                    ignore_requires_python=False,
-                    ignore_installed=True,
-                    isolated=False,
-                    wheel_cache=self.wheel_cache,
-                    use_user_site=False,
-                )
-                self.resolver.resolve(reqset)
-                self._dependencies_cache[ireq] = reqset.requirements.values()
-            reqset.cleanup_files()
-        return set(self._dependencies_cache[ireq])
+                self._dependencies_cache[ireq] = self.resolve_reqs(download_dir, ireq, wheel_cache)
+            finally:
+                if 'PIP_REQ_TRACKER' in os.environ:
+                    if prev_tracker:
+                        os.environ['PIP_REQ_TRACKER'] = prev_tracker
+                    else:
+                        del os.environ['PIP_REQ_TRACKER']
+                try:
+                    self.wheel_cache.cleanup()
```

Comment:
@techalchemy shouldn't there be a `wheel_cache.cleanup()` instead?

Reply:
Good catch! And how unfortunate that the error is masked by the try-except. Maybe instead of try-except, it would be better to use:
```python
if hasattr(wheel_cache, 'cleanup'):
    wheel_cache.cleanup()
```
IMHO that's easier to read and less error prone.

Reply:
The `hasattr()` has a quircks on Python 2, see the explanation: https://hynek.me/articles/hasattr/

Take a look at my suggestion #968, btw.